src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1323,6 +1324,55 @@ public async Task TestEngineCanStartConfigWithCustomProperties()
             }
         }
 
+        /// <summary>
+        /// This test checks that the GetJsonSchema method of the JsonConfigSchemaValidator class
+        /// correctly downloads a JSON schema from a given URL, and that the downloaded schema matches the expected schema.
+        /// </summary>
+        [TestMethod]
+        public async Task GetJsonSchema_DownloadsSchemaFromUrl()
+        {
+            // Arrange
+            Mock<HttpMessageHandler> handlerMock = new(MockBehavior.Strict);
+            string jsonSchemaContent = "{\"type\": \"object\", \"properties\": {\"property1\": {\"type\": \"string\"}}}";
```

Comment:
It sounds like the schema validation was never really working, but now that we use our own http client to get get from the url, it should function. However, if we are mocking the value returned, are we really testing the case that was failing previously? Is there a way to actually get the file that we want, or perhaps have it available locally, so that the test validates we have fixed the scenario?

Reply:
it was working, we have a fail safe method, if it was not able to fetch from the url due to any reasons, it gets the schema from the package itself to verify. 
We were using a different method before and then casting it to string, it was failing before because some refrences were not resolved when reading it as JsonSchema, since we need the data as string, this new approach is much better because it gives the data in the format we need.

This test here is to verify that we are able to fetch the data from the url.

Reply:
> it was failing before because some refrences were not resolved when reading it as JsonSchema, 

This wasn't in pr description. can you elaborate since this sounds like it is more than just URL fetching that was an issue, and instead, was an issue with how the fetched JSON schema was processed by newtonsoft. Do unresolved references mean that there is something wrong with our JSON schema?

Reply:
> it was working, we have a fail safe method, if it was not able to fetch from the url due to any reasons, it gets the schema from the package itself to verify. We were using a different method before and then casting it to string, it was failing before because some refrences were not resolved when reading it as JsonSchema, since we need the data as string, this new approach is much better because it gives the data in the format we need.
> 
> This test here is to verify that we are able to fetch the data from the url.

So if we were to have this same test with the mocked data being returned, but with the previous code from before the fix, would it have passed?

My concern is just that we dont have a test that would fail before and pass now to prevent regression, if that makes sense.

Reply:
we didn't had a test before for testing the url section, the previous tests were passing because all of them were fetching the schema from the package and not the url. this time i added a test that tests the url. So now we have tests that covers both scenarios.